controller.go

```diff
@@ -269,9 +269,14 @@ func (c *Controller) syncHandler(key string) error {
 
 		// Trigger a manual Job for the initial sync
 		_, err = c.kubeclientset.BatchV1().Jobs(apprepo.Namespace).Create(newJob(apprepo))
-	} else if shouldUpdate(cronjob, apprepo) {
-		glog.V(4).Infof("Updating CronJob %q for AppRepository %q", cronjobName, apprepo.GetName())
-		cronjob, err = c.kubeclientset.BatchV1beta1().CronJobs(apprepo.Namespace).Update(newCronJob(apprepo))
+	} else {
+		if shouldUpdate(cronjob, apprepo) {
+			glog.V(4).Infof("Updating CronJob %q for AppRepository %q", cronjobName, apprepo.GetName())
+			cronjob, err = c.kubeclientset.BatchV1beta1().CronJobs(apprepo.Namespace).Update(newCronJob(apprepo))
+		}
+
+		// The AppRepository has changed, launch a manual Job
+		_, err = c.kubeclientset.BatchV1().Jobs(apprepo.Namespace).Create(newJob(apprepo))
```

Comment:
I think this should be a bit smarter and only launch a manual job if the counter or url changed. We may need to move the `shouldUpdate` up to the event listener.

Reply:
the `shouldUpdate` function is meant to update the cronJob (because the URL has changed). In this case we want to launch the single job both if the URL has changed or if the counter has been increased. We can try to get the previous specification to discover that.